agent/exec/controller.go

```diff
@@ -104,17 +104,21 @@ func Resolve(ctx context.Context, task *api.Task, executor Executor) (Controller
 
 	// depending on the tasks state, a failed controller resolution has varying
 	// impact. The following expresses that impact.
-	if task.Status.State < api.TaskStateStarting {
-		if err != nil {
-			// before the task has been started, we consider it a rejection.
-			status.Message = "resolving controller failed"
-			status.Err = err.Error()
+	if err != nil {
```

Comment:
Not exactly sure on how this works - what happens if the error is **not** `nil` (the controller can be resolved), but `task.Status.State < api.TaskStateStarting`?

It looks like we won't trigger the code below.

Is that true?